ios/engine/KMEI/KeymanEngine/Classes/Extension/UserDefaults+Types.swift

```diff
@@ -133,6 +133,20 @@ public extension UserDefaults {
       set(lexicalModels, forKey: Key.userLexicalModelsList)
     }
   }
+
+  var userResources: [LanguageResource]? {
+    get {
+      let keyboards = userKeyboards ?? []
+      let lexicalModels = userLexicalModels ?? []
+      
+      let resources: [LanguageResource] = keyboards + lexicalModels
+      if resources.count == 0 {
+        return nil
```

Comment:
I'm curious why you don't just return an empty array here? After all, in the [only place this seems to be used](https://github.com/keymanapp/keyman/pull/2007/files#diff-b39e38886315bc8b5ff31bc65126e3c8R95), you just undo the nil and turn it into an empty array again :)

Reply:
I've got a _lot_ of ideas for tech debt retirement in v13 that could make great use of this new property.  I wanted its output to mirror the properties of `keyboards` and `lexicalModels`, as it's basically an abstracted union of the two - if both are `nil`, it will probably need to be `nil` at some point during the tech-debt retirement process.